k-distribution/include/builtin/domains.k

```diff
@@ -155,6 +155,23 @@ module MAP
   syntax KItem ::= choice(Map)                      [function, hook(MAP.choice), klabel(Map:choice)]
 endmodule
 
+module MAP-SYMBOLIC [symbolic]
+  imports MAP
+  imports K-EQUAL
+
+  rule ((K1 |-> V1) MAP) [ K2 ] => V1         requires K1  ==K K2
+  rule ((K1 |-> V1) MAP) [ K2 ] => MAP [ K2 ] requires K1 =/=K K2
+
+  rule ([ K1 <- K2 ] MAP) [ K2 ] => V1         requires K1  ==K K2
+  rule ([ K1 <- K2 ] MAP) [ K2 ] => MAP [ K2 ] requires K1 =/=K K2
+
+  rule ((K1 |-> V1) MAP) [ K2 <- V2 ] => (K1 |-> V2) MAP                requires K1  ==K K2
+  rule ((K1 |-> V1) MAP) [ K2 <- V2 ] => (K1 |-> V1) (MAP [ K2 <- V2 ]) requires K1 =/=K K2
```

Comment:
so, you want to keep the new entry in the innermost?

Reply:
I've commented out this rule because it could infinite loop.